src/firstboot/firstboot.c

```diff
@@ -604,6 +604,22 @@ static int prompt_root_password(void) {
         return 0;
 }
 
+static int find_shell(const char *path, const char *root)
+{
+        int r;
+
+        assert(path);
+
+        if (!valid_shell(path))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "%s is not a valid shell", path);
+
+        r = chase_symlinks(path, root, CHASE_PREFIX_ROOT, NULL, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to resolve shell %s: %m", prefix_roota(root, path));
```

Comment:
alloca cannot be called in arguments to a function. So a temp variable needs to be declared, and alloca called, and then the variable can be used in the function call argument.